lib/models/application.coffee

```diff
@@ -237,24 +257,25 @@ getApplicationModel = (deps, opts) ->
 	# @function
 	# @memberof resin.models.application
 	#
-	# @param {String} name - application name
+	# @param {String|Number} nameOrId - application name (string) or id (number)
 	# @returns {Promise}
 	#
 	# @example
 	# resin.models.application.remove('MyApp');
 	#
 	# @example
+	# resin.models.application.remove(123);
+	#
+	# @example
 	# resin.models.application.remove('MyApp', function(error) {
 	# 	if (error) throw error;
 	# });
 	###
-	exports.remove = (name, callback) ->
-		exports.get(name).then ->
+	exports.remove = (nameOrId, callback) ->
+		exports.get(nameOrId).then (application) ->
```

Comment:
Another note: to reduce the load on the API and the amount of data flying over the network we should add support for extra PineJS options (mentioned already in a different thread) and in this specific case pass `$select: 'id'` which will return an object with ID only.
We should intro this optimization to other places that only need the specific props of the resource.
Of course this can go into a separate PR that will implement the custom props in the first place.